### PR TITLE
Payments Blocks; remove thickbox

### LINKED ISSUE #71048: Rework the checkout modal
We use `thickbox` library to display the modal iframe:
https://github.com/Automattic/jetpack/blob/b313bd253afdf919adf7a71a866d0ab85b3a7bda/projects/plugins/jetpack/extensions/shared/memberships.js#L19

That has a dependancy on jQuery, which is less than ideal. We want to remove that dependency and switch to another modal system.
Suposedly, navigation modal has a good library:

https://make.wordpress.org/core/2022/01/07/the-new-navigation-block/

**Remember that checkout modal displays for the site visitor, not in Gutenberg editor.** This means view.js in gutenberg components.

### Goal of this task:

1. Figure out how to use the navigation block modal to display an iframe. If that is not possible, research other Gutenberg components and see if there is a modal that fits.
2. Reimplement checkout using that new modal.

Watch out for:
- Mobile layouts
- Cross browser compatibilities

iOS safari MUST WORK FLAWLESSLY.

CC @mtias

### PR DESCRIPTION
**Pull request by n3f**
Fixes https://github.com/Automattic/wp-calypso/issues/71048.  Requires D95805-code to be merged first.

#### Changes proposed in this Pull Request:

*

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Testing instructions:

* Go to '..'
*

### CODE REVIEW

**Comment by millerf** [comment]
I think it needs to be removed [from here](https://github.com/Automattic/jetpack/pull/26417/files#diff-9164d71f603706b3c6cc58c71ba3f2fbb12b1dba772adca76793f60c24ddbca7R356-R359) as well

**Comment by n3f** [comment]
> I think it needs to be removed [from here](https://github.com/Automattic/jetpack/pull/26417/files#diff-9164d71f603706b3c6cc58c71ba3f2fbb12b1dba772adca76793f60c24ddbca7R356-R359) as well

Odd, my copy of subscriptions.php doesn't have that line.  I'm going to have to investigate.

**Comment by n3f** [comment]
> I think it needs to be removed [from here](https://github.com/Automattic/jetpack/pull/26417/files#diff-9164d71f603706b3c6cc58c71ba3f2fbb12b1dba772adca76793f60c24ddbca7R356-R359) as well

🤦 , it's because that's new code added with the PR.  I'm guessing this will need to be rebased after the paid newsletter PR gets merged...  Or @artpi, if this functionality gets far enough should I look at merging this with the paid-newsletter branch first?

**Comment by millerf** [comment]
I think the scrollbar should definitely be on the whole page and not in the modal...


| With thickbox | Without thickbox |
|----|----|
| Screenshot 2023-01-10 at 15 55 40 |  Screenshot 2023-01-10 at 15 55 19 |

**Comment by n3f** [comment]
> I think the scrollbar should definitely be on the whole page and not in the modal...

Thanks for looking at this! I'll see what I can do 😆

File: composer.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
 		"php-parallel-lint/php-parallel-lint": "1.3.2",
 		"sirbrillig/phpcs-changed": "2.10.0",
 		"squizlabs/php_codesniffer": "^3.6.2",
-		"wp-coding-standards/wpcs": "2022-02-07 as 2.3.1"
+		"wp-coding-standards/wpcs": "2023-01-04 as 2.3.1"
```
Comment (artpi) [inline composer.json]: Make sure this is something you want to do

Comment (n3f) [reply composer.json]: Git pull did something weird.  I did a manual rebase and it fixed it.

**Comment by artpi** [comment]
I am voting to put this temporarily on hold and ship https://github.com/Automattic/jetpack/pull/28318 for now.
We will probably need a more comprehensive rewrite of the checkout and then we can continue this work

File: projects/plugins/jetpack/extensions/shared/memberships-modal.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+			{ isOpen && (
+				<Modal
+					ref={ modalRef }
+					onRequestClose={ closeModal }
+					shouldCloseOnClickOutside={ false }
+					__experimentalHideHeader={ true }
+					// isFullScreen={ true }
+					className="jetpack-memberships-modal"
+				>
+					<div class="lds-ring">
```
Comment (n3f) [inline projects/plugins/jetpack/extensions/shared/memberships-modal.js:78]: Use `spinner` component instead.

File: projects/plugins/jetpack/extensions/shared/memberships-modal.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+						<div></div>
+						<div></div>
+						<div></div>
+						<div></div>
+					</div>
```
Comment (n3f) [inline projects/plugins/jetpack/extensions/shared/memberships-modal.js:83]: Use `spinner` component instead.

**Comment by n3f** [comment]
@wordpress elements are only meant for editor and not the frontend.

**Pull request closed without merging**